src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/clients/AsylumCaseFeeApiPayment.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.iacaseapi.infrastructure.clients;
+
+import static java.util.Objects.requireNonNull;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.service.FeePayment;
+
+@Service
+public class AsylumCaseFeeApiPayment implements FeePayment<AsylumCase> {
```

Comment:
This class probably belongs in the `domain` package since it is not actually the client which calls the `ia-case-payments-api`?  That would be the delegator.

Also maybe this could be named better since it's a service class?

Reply:
Moved to service and renamed to AsylumCaseFeePaymentService

Reply:
And renamed and moved the corresponding test

Reply:
👍